redbeams/src/main/java/com/sequenceiq/redbeams/service/stack/RedbeamsUpgradeService.java

```diff
@@ -46,6 +64,57 @@ public class RedbeamsUpgradeService {
     @Inject
     private OperationService operationService;
 
+    @Inject
+    private CredentialService credentialService;
+
+    @Inject
+    private CredentialToCloudCredentialConverter credentialConverter;
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private PersistenceNotifier persistenceNotifier;
+
+    @Inject
+    private DBStackToDatabaseStackConverter databaseStackConverter;
+
+    public UpgradeDatabaseResponse validateUpgradeDatabaseServer(String crn, UpgradeDatabaseRequest upgradeDatabaseRequest) throws Exception {
+        DBStack dbStack = dbStackService.getByCrn(crn);
+        MDCBuilder.buildMdcContext(dbStack);
+        MDCBuilder.addEnvironmentCrn(dbStack.getEnvironmentId());
+
+        MajorVersion currentVersion = dbStack.getMajorVersion();
+        TargetMajorVersion targetVersion = upgradeDatabaseRequest.getTargetMajorVersion();
+
+        LOGGER.debug("Validate upgrade called for: {}, with target version: {}, current version is: {}", dbStack, targetVersion, currentVersion);
+        Location location = location(region(dbStack.getRegion()), availabilityZone(dbStack.getAvailabilityZone()));
+        String accountId = dbStack.getAccountId();
+        CloudContext cloudContext = CloudContext.Builder.builder()
+                .withId(dbStack.getId())
+                .withName(dbStack.getName())
+                .withCrn(dbStack.getResourceCrn())
+                .withPlatform(dbStack.getCloudPlatform())
+                .withVariant(dbStack.getPlatformVariant())
+                .withLocation(location)
+                .withUserName(dbStack.getUserName())
+                .withAccountId(accountId)
+                .build();
+        Credential credential = credentialService.getCredentialByEnvCrn(dbStack.getEnvironmentId());
+        CloudCredential cloudCredential = credentialConverter.convert(credential);
+        CloudConnector connector = cloudPlatformConnectors.get(cloudContext.getPlatformVariant());
+        AuthenticatedContext ac = connector.authentication().authenticate(cloudContext, cloudCredential);
+        DatabaseStack databaseStack = databaseStackConverter.convert(dbStack);
+
+        ResourceConnector resourceConnector = connector.resources();
+        try {
+            resourceConnector.validateUpgradeDatabaseServer(ac, databaseStack, persistenceNotifier, targetVersion);
+            return new UpgradeDatabaseResponse(null, null, dbStack.getMajorVersion());
+        } catch (CloudConnectorException ex) {
```

Comment:
my suggestion is to catch every exception here coming from the provider, so it would be translated into validation failure (eg aws api not reachable or missing permission for describing) instead of being an internal server error